scrapers_next/pa/committees.py

```diff
@@ -0,0 +1,171 @@
+    example_name = "Aging & Older Adult Services"
+    example_input = ScrapeCommittee(
+        name=example_name, classification="committee", chamber="lower"
+    )
+
+    def process_page(self):
+        com = self.input
+        try:
+            # This section has the chair memebers the regular, democratic and minority and the roles
+            # main chair
+            Chair_Member = (
+                CSS("div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText a")
+                .match(self.root)[0]
+                .text.strip()
+            )
+            # main chair role
+            Chair_Member_role = (
+                CSS(
+                    "div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText div"
+                )
+                .match(self.root)[0]
+                .text.strip()
+            )
+            com.add_member(Chair_Member, Chair_Member_role)
+            # Democratic Chair member and or the minority chair member
+            Demo_Chair_Member = (
+                CSS("div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText a")
+                .match(self.root)[1]
+                .text.strip()
+            )
+            # Democratic Chair member and or the minority chair member role
+            Demo_Chair_Member_role = (
+                CSS(
+                    "div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText div"
+                )
+                .match(self.root)[1]
+                .text.strip()
+            )
+            com.add_member(Demo_Chair_Member, Demo_Chair_Member_role)
+        except IndexError:
+            pass
+        Majority_Members = CSS(
+            ".Widget.CteeInfo-MajorityList .MemberInfoList-MemberWrapper.Member"
+        ).match(self.root)
+        for mem in Majority_Members:
+            try:
+                major_member_name = CSS("div a").match_one(mem).text.strip()
+                major_mem_position = CSS(".position").match_one(mem).text.strip()
+            except SelectorError:
+                major_mem_position = "Member"
+            com.add_member(major_member_name, major_mem_position)
+        Minority_Members = CSS(
+            ".Widget.CteeInfo-MinorityList .MemberInfoList-MemberWrapper.Member"
+        ).match(self.root)
+        for mem in Minority_Members:
+            try:
+                minor_member_name = CSS("div a").match_one(mem).text.strip()
+                minor_mem_position = CSS(".position").match_one(mem).text.strip()
+            except SelectorError:
+                minor_mem_position = "Member"
+            com.add_member(minor_member_name, minor_mem_position)
+        return com
+
+
+class SenateCommitteeDetail(HtmlPage):
+    source = "https://www.legis.state.pa.us/cfdocs/CteeInfo/index.cfm?Code=32&CteeBody=H&SessYear=2021"
+    input = "Aging & Older Adult Services"
+
+    def process_page(self):
```

Comment:
Is the code in `SenateCommitteeDetail`'s `process_page` method exactly the same as `HouseCommitteeDetail`'s `process_page` method? It looks like it is. There are times when it makes sense to repeat a few lines of code in multiple places, however repeating two long functions that are exactly the same is a risky code pattern. Someone could find a bug in one and not know that they need to fix the other. It's not obvious to a later developer that these are exactly the same segments of code (they are long, which makes it harder to compare). If the HTML structure of the Senate and House committee pages are fundamentally the same, then it makes sense to use a single function to process both.

You can define a function outside of the classes, that can be called inside of the `process_page` method in both classes, see [this example](https://github.com/openstates/openstates-scrapers/blob/455cf66c6460a819a4e7f8bd31df59a5c7d4016e/scrapers_next/fl/people.py#L7).